### PR TITLE
Add test settings view

### PR DESCRIPTION
**Pull request by ziyiz-amzn**
### Description of changes:
Add a new view in the demo app to allow to override the meeting endpoint url

### Testing done:

1. Join a meeting without changing meeting endpoint url
2. Join a meeting with an overridden meeting endpoint url
3. Join a meeting with an overridden url which contains leading or trailing spaces
4. Join a meeting with an empty overridden url
5. Join a meeting with an url without suffix "/"

#### Manual test cases (add more as needed):

- [x] Join meeting without CallKit
- [ ] Join meeting with CallKit as Incoming
- [ ] Join meeting with CallKit as Outgoing
- [x] Leave meeting
- [ ] Rejoin meeting
- [ ] See metrics
- [ ] See attendee presence
- [ ] Send audio
- [ ] Receive audio
- [ ] Mute self
- [ ] Unmute self
- [ ] See local mute indicator when muted
- [ ] See remote mute indicator when other is muted
- [ ] Enable and disable local video
- [ ] Enable and disable remote video from remote side
- [ ] Send local video
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Receive remote screen share

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [x] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:
debug_settings_landscape
debug_settings_portrait
joining_landscape
joining_portrait

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/JoiningViewController.swift [hunk 1]
```diff
@@ -18,8 +18,11 @@ class JoiningViewController: UIViewController, UITextFieldDelegate {
     @IBOutlet var joinWithoutCallKitButton: UIButton!
     @IBOutlet var joinAsIncomingCallButton: UIButton!
     @IBOutlet var joinAsOutgoingCallButton: UIButton!
+    @IBOutlet var DebugSettingsButton: UIButton!
 
     private let toastDisplayDuration = 2.0
+    private let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+    var model: DebugSettingsModel = DebugSettingsModel()
```
Comment (zhinang-amazon) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/JoiningViewController.swift:25]: Nit: debugSettingsModel

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/JoiningViewController.swift [hunk 2]
```diff
@@ -72,7 +84,7 @@ class JoiningViewController: UIViewController, UITextFieldDelegate {
             return
         }
 
-        MeetingModule.shared().prepareMeeting(meetingId: meetingId, selfName: name, option: callKitOption) { success in
+        MeetingModule.shared().prepareMeeting(meetingId: meetingId, selfName: name, option: callKitOption, overriddenEndpoint: model.endpointUrl) { success in
```
Comment (zhinang-amazon) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/JoiningViewController.swift:87]: Nit: line is too long, break the params into multiple lines, same for the two changes in MeetingModule

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/DebugSettingsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class DebugSettingsViewController: UIViewController, UITextFieldDelegate {
+    @IBOutlet var serverEndpointUrlTextField: UITextField!
+    @IBOutlet var saveButton: UIButton!
+    
+    var model: DebugSettingsModel?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.serverEndpointUrlTextField.delegate = self
+        self.serverEndpointUrlTextField.text = model?.endpointUrl
```
Comment (zhinang-amazon) [inline AmazonChimeSDKDemo/AmazonChimeSDKDemo/DebugSettingsViewController.swift:20]: Nit: there is no need for `self.` in most cases, unless it's in a closure.

File: AmazonChimeSDKDemo/AmazonChimeSDKDemo/JoiningViewController.swift (hunk 1 above)
Comment (ziyiz-amzn) [reply AmazonChimeSDKDemo/AmazonChimeSDKDemo/JoiningViewController.swift:25]: Done

**Review by hokyungh — APPROVED** [review]

**Review by zhinang-amazon — DISMISSED** [review]

**Review by hokyungh — APPROVED** [review]

**Review by zhinang-amazon — APPROVED** [review]

**Pull request merged**